OpenRA.Mods.Common/Warheads/SpreadDamageWarhead.cs

```diff
@@ -60,10 +64,28 @@ public override void DoImpact(WPos pos, Actor firedBy, WarheadArgs args)
 					.Select(s => Pair.New(s, s.DistanceFromEdge(victim, pos)))
 					.MinByOrDefault(s => s.Second);
 
-				// Cannot be damaged without an active HitShape
+				// Cannot be damaged without an active HitShape.
 				if (closestActiveShape.First == null)
 					continue;
 
+				var falloffDistance = 0;
+				switch (DamageCalculationType)
+				{
+					case DamageCalculationType.HitShape:
+						falloffDistance = closestActiveShape.Second.Length;
+						break;
+					case DamageCalculationType.ClosestTargetablePosition:
+						falloffDistance = victim.GetTargetablePositions().Select(x => (x - pos).Length).Min();
+						break;
+					case DamageCalculationType.CenterPosition:
+						falloffDistance = (victim.CenterPosition - pos).Length;
+						break;
+				}
+
+				// The range to target is more than the range the warhead covers, so GetDamageFalloff() is going to give us 0 and we're going to do 0 damage anyway, so bail early.
+				if (falloffDistance > Range[Range.Length - 1].Length)
+					continue;
+
 				var localModifiers = args.DamageModifiers.Append(GetDamageFalloff(closestActiveShape.Second.Length));
```

Comment:
Not 100% sure, but wouldn't this need to be `GetDamageFalloff(falloffDistance)` then?

Reply:
Derp doesn't even begin to cover it.

Reply:
Fixed.